### DISCUSSION TITLE
Force livewire asset injection

### DISCUSSION THREAD

**Discussion by ZARk-be** [filamentphp/filament#11235] [Ideas]
Hello

If livewire is configured to not inject assets

**config/livewire.php** `inject_assets = false`

Then filament will not render.

It's possible to add` \Livewire\Livewire::forceAssetInjection();`  in the filament routes to compensate this.
Maybe an option or a check ? 
Took us a while to figure the culprit of why filament was not rendering.

**Comment by abdelhamiderrahmouni** [comment]
here is a temporary fix 
```php
// AppServiceProvider 
if(Filament::getCurrentPanel()) {
    Config::set('livewire.inject_assets', true);
}
```

Credits to gbuckingham89 @ [Including livewire.js when livewire.inject_assets = false](https://www.answeroverflow.com/m/1182031415227985931)

**Reply by DanielSpravtsev** [reply]
This bug is hard to debug! :) Thanks

**Reply by vlados** [reply]
`Filament::getCurrentPanel` is returning Panel. It's nothing to do with check of the request. The check should be
```php
if (request()->is(Filament::getCurrentPanel()->getId().'*')) {
...
}
```

**Reply by orzcc** [reply]
The getId is also inappropriate, and errors will occur when changing the path. The reasonable approach should be:

```
if (request()->is(Filament::getCurrentPanel()->getPath().'*')) {
    Config::set('livewire.inject_assets', true);
}
```

**Comment by camerontucker** [comment]
Worked until I added Octane, which has an empty request() in AppServiceProvider.

However! Luckily ChatGPT knows all things so just put it in middleware and stick it onto AdminPanelProvider.

```
class InjectLivewire
{
    public function handle(Request $request, Closure $next): Response
    {
        Config::set('livewire.inject_assets', true);

        return $next($request);
    }
}
```

```
class AdminPanelProvider extends PanelProvider
{
    public function panel(Panel $panel): Panel
    {
        return $panel
            ...
            ->middleware([
                ...
                InjectLivewire::class
            ])...
```
